Providers/Profile.swift

```diff
@@ -1082,33 +1099,54 @@ public class BrowserProfile: Profile {
         }
 
         private func syncClientsWithDelegate(delegate: SyncDelegate, prefs: Prefs, ready: Ready) -> SyncResult {
-            log.debug("Syncing clients to storage.")
-            let clientSynchronizer = ready.synchronizer(ClientsSynchronizer.self, delegate: delegate, prefs: prefs)
-            return clientSynchronizer.synchronizeLocalClients(self.profile.remoteClientsAndTabs, withServer: ready.client, info: ready.info)
+            return statsWrapEngineSync("clients") { statsDelegate in
```

Comment:
I wasn't sure if I wanted to conflate the two since it wasn't clear what SyncDelegate was around for. Moving the stats stuff into it will clean this PR up quite a bit.